### PR TITLE
Add audit multiple directory support

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/commands/audit/generic/auditmanager.go [hunk 1]
```diff
@@ -21,8 +22,54 @@ import (
...
+	serverDetails *config.ServerDetails,
+	excludeTestDeps,
+	useWrapper,
+	insecureTls bool,
+	args []string,
+	progress ioUtils.ProgressMgr,
+	requirementsFile string,
+	ignoreConfigFile bool,
+	workingDirs []string,
+	technologies ...string) (results []services.ScanResponse, isMultipleRoot bool, err error) {
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:37]: Let's add a new-line after this line, to make it easier to see where the function's body starts.

File: xray/commands/audit/generic/auditmanager.go [hunk 2]
```diff
@@ -21,8 +22,54 @@ import (
...
+	if len(workingDirs) == 0 {
+		return singleGenericAudit(xrayGraphScanParams, serverDetails, excludeTestDeps, useWrapper, insecureTls, args, progress, requirementsFile, ignoreConfigFile, technologies...)
+	}
+	projectDir, err := os.Getwd()
+	if errorutils.CheckError(err) != nil {
+		return
+	}
+	var errorList []string
+	defer os.Chdir(projectDir)
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:46]: Let's not ignore the error returned by the `Chdir` function.

File: xray/commands/audit/generic/auditmanager.go [hunk 3]
```diff
@@ -21,8 +22,54 @@ import (
...
+	for _, wd := range workingDirs {
+		absWd := filepath.Join(projectDir, wd)
+		log.Info("Audit project: " + absWd)
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:49]: Audit --> Auditing

File: xray/commands/audit/generic/auditmanager.go [hunk 4]
```diff
@@ -21,8 +22,54 @@ import (
...
+		e := os.Chdir(absWd)
+		if e != nil {
+			// Save the error but continue to the other paths
+			errorList = append(errorList, fmt.Sprintf("audit command couldn't find the following path:%s\n%s", absWd, e.Error()))
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:53]: audit command... --> the audit command...
...path:%s\n%s --> ...path: %s\n%s

File: xray/commands/audit/generic/auditmanager.go [hunk 5]
```diff
@@ -21,8 +22,54 @@ import (
...
+		}
+	}
+	if len(errorList) > 0 {
+		err = errors.New(strings.Join(errorList, "\n"))
+	}
+	return
+}
+
+// GenericAudit audits the project found in the current directory using Xray.
+func singleGenericAudit(
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:72]: singleGenericAudit --> doAudit

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: xray/commands/audit/generic/auditmanager.go [hunk 6]
```diff
@@ -21,8 +22,54 @@ import (
...
+		return singleGenericAudit(xrayGraphScanParams, serverDetails, excludeTestDeps, useWrapper, insecureTls, args, progress, requirementsFile, ignoreConfigFile, technologies...)
+	}
+	projectDir, err := os.Getwd()
+	if errorutils.CheckError(err) != nil {
+		return
+	}
+	var errorList []string
+	defer os.Chdir(projectDir)
+	for _, wd := range workingDirs {
+		absWd := filepath.Join(projectDir, wd)
```
Comment (omerzi) [inline xray/commands/audit/generic/auditmanager.go:48]: Maybe we should add a sanity check to ensure the working dir path isn't already provided by the user as a parameter? For example, In the Frogbot context, inside 'downloadRepoToTempDir', we already set the whole path and send it to GenericAudit.

**Pull request merged**